pkg/dataobj/logs_reader.go

```diff
@@ -296,6 +296,21 @@ func translateLogsPredicate(p LogsPredicate, columns []dataset.Column, columnDes
 		}
 		return convertLogsTimePredicate(p, timeColumn)
 
+	case LogMessageFilterPredicate:
+		messageColumn := findColumnFromDesc(columns, columnDesc, func(desc *logsmd.ColumnDesc) bool {
+			return desc.Type == logsmd.COLUMN_TYPE_MESSAGE
+		})
+		if messageColumn == nil {
+			return dataset.FalsePredicate{}
+		}
+
+		return dataset.FuncPredicate{
+			Column: messageColumn,
+			Keep: func(_ dataset.Column, value dataset.Value) bool {
+				return p.Keep(valueToString(value))
```

Comment:
we can just the byteslice once this is merged https://github.com/grafana/loki/pull/16747

Reply:
merged!

Reply:
nice, thanks!